modules/rules/main.tf

```diff
@@ -0,0 +1,96 @@
+##############################################################################
+# SCC rules
+##############################################################################
+
+resource "ibm_scc_rule" "scc_rule_instance" {
+  count       = length(var.rules) > 0 ? length(var.rules) : 0
```

Comment:
NOTE: am not suggesting a change here unless there is something we can use in the map for a for_each!

Using count for these types of resources can be disruptive, as changes to the array itself can cause "destroy/create" for many objects that do not actually have changes. For instance, a rule is removed from middle of array will cause destroy/create for all rules that came after that rule, due to the index changing for everything after the removed rule (same for add).

In this case I don't think it is as big of issue because rules getting destroyed and recreated shouldn't have much of an impact.

Also if using a "for_each" logic, there needs to be something in the map that is unique for each entry, and I don't see anything in the map that could be used for that.

Reply:
hmm @toddgiguere rules will be used as part of a control library, so if rules get destroyed and recreated they will get created with a new ID, which will then break the control library. So I think we may need to figure out how to prevent that?

Reply:
@jor2 @ocofaigh Are we in agreement then that we should use a for_each here instead of a count so that we are not relying on the positional index of the incoming rules array?

Reply:
yes I think so - I guess we can confirm with a quick test if we wanted